app/controllers/search_controller.rb

```diff
@@ -0,0 +1,28 @@
+class SearchController < ApplicationController
+  skip_authorization_check
+  
+  respond_to :html, :json, :js
+
+  def index
+    @tagged_products = Product.includes(:reviews).tagged_with(params[:query])
+    @product_results = @tagged_products + PgSearch.multisearch(params[:query]).includes(:searchable).where(searchable_type: 'Product').map(&:searchable)
+    @product_results = Kaminari.paginate_array(@product_results).page(params[:product]).per(4)
+
+    @article_results = PgSearch.multisearch(params[:query]).includes(:searchable).where(searchable_type: 'Article').map(&:searchable)
+    @article_results = Kaminari.paginate_array(@article_results).page(params[:article]).per(4)
+
+    @reference_results = PgSearch.multisearch(params[:query]).includes(:searchable).where(searchable_type: 'Reference').map(&:searchable)
+    @reference_results = Kaminari.paginate_array(@reference_results).page(params[:resource]).per(4)
+
+    @tag_results = PgSearch.multisearch(params[:query]).includes(:searchable).where(searchable_type: 'Tag').map(&:searchable)
+    @tag_results = Kaminari.paginate_array(@tag_results).page(params[:tag]).per(4)
```

Comment:
@cchanningallen Thanks! I took your advice, but still need to use `.map` because when I tried to use `.pluck` I got this error:

Cannot eagerly load the polymorphic association :searchable

I don't know anything about polymorphic associations, so that has to be from the way the PgSearch gem is setup.